qiskit/visualization/matplotlib.py

```diff
@@ -342,9 +342,16 @@ def _get_colors(self, op):
             ec = self._style.edge_color
             lc = self._style.lc
         if op.name == 'reset':
+            gt = self._style.lc
+        else:
+            gt = self._style.gt
+        if op.name not in {'h', 't', 's', 'z', 'sdg', 'tdg', 'u1', 'reset', 'meas', 'measure'}:
             gt = self._style.not_gate_lc
+            self._style.sc = self._style.not_gate_lc
         else:
             gt = self._style.gt
+            self._style.sc = self._style.lc
```

Comment:
The text colors should not be hardcoded in `matplotlib.py` but should be a dictionary in the Style class like the gate color. Otherwise we won't be able to use different style sheets that use different combinations of gate and text color.

Reply:
Understood. I'll fix it